x-pack/auditbeat/module/system/host/users_unix.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build !windows
+
+package host
+
+// #include <sys/types.h>
+// #include <pwd.h>
+import "C"
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// User represents a user. Fields according to getpwent(3).
+type User struct {
+	name   string
+	passwd string
+	uid    uint32
+	gid    uint32
+	gecos  string
+	dir    string
+	shell  string
+}
+
+func (user User) toMapStr() common.MapStr {
+	return common.MapStr{
+		"name":   user.name,
+		"passwd": user.passwd,
+		"uid":    user.uid,
+		"gid":    user.gid,
+		"gecos":  user.gecos,
```

Comment:
Can some of these values be empty strings? I was thinking that some of them are not always populated so you should drop those keys.